### ISSUE TITLE
doctor: Cannot parse non-semver revisions

### ISSUE THREAD

**Issue by uhthomas** [twpayne/chezmoi#1607]
## Describe the bug

The `chezmoi doctor` command marks a check as `failed` when using a tool which doesn't report its version as semver.

```sh
❯ chezmoi doctor
RESULT    CHECK                MESSAGE
...
failed    age-command          found /usr/bin/age, could not parse version from (devel)
...
```

## To reproduce

Install `age` from the official [ABS](https://wiki.archlinux.org/title/Arch_Build_System) community repository: [community/age](https://archlinux.org/packages/community/x86_64/age/).

Verify `age` is reporting its version as `devel` (this is weird tbh).

```sh
❯ age --version
(devel)
```

Observe `chezmoi doctor` produce a `failed` result for the `age-command` check.

## Expected behavior

The result should be `ok` and the version should be `devel`.

## Output of command with the `--verbose` flag

```sh
❯ chezmoi doctor -v
RESULT    CHECK                MESSAGE
warning   version              v2.7.5, built at 2021-11-08T11:20:41Z
ok        os-arch              linux/amd64 (Arch Linux)
ok        executable           /usr/bin/chezmoi
ok        config-file          ~/.config/chezmoi/chezmoi.toml
ok        source-dir           ~/.local/share/chezmoi is a directory
ok        suspicious-entries   no suspicious entries
ok        working-tree         ~/.local/share/chezmoi is a directory
ok        dest-dir             ~ is a directory
ok        shell                found /usr/bin/zsh
ok        edit-command         found /usr/bin/vim
ok        umask                022
ok        git-command          found /usr/bin/git, version 2.33.1
ok        merge-command        found /usr/bin/vimdiff
failed    age-command          found /usr/bin/age, could not parse version from (devel)
ok        gpg-command          found /usr/bin/gpg, version 2.2.29
info      pinentry-command     not set
info      1password-command    op not found in $PATH
info      bitwarden-command    bw not found in $PATH
info      gopass-command       gopass not found in $PATH
info      keepassxc-command    keepassxc-cli not found in $PATH
info      keepassxc-db         not set
info      lastpass-command     lpass not found in $PATH
info      pass-command         pass not found in $PATH
info      vault-command        vault not found in $PATH
info      secret-command       not set
```

## Output of `chezmoi doctor`

```sh
❯ chezmoi doctor
RESULT    CHECK                MESSAGE
warning   version              v2.7.5, built at 2021-11-08T11:20:41Z
ok        os-arch              linux/amd64 (Arch Linux)
ok        executable           /usr/bin/chezmoi
ok        config-file          ~/.config/chezmoi/chezmoi.toml
ok        source-dir           ~/.local/share/chezmoi is a directory
ok        suspicious-entries   no suspicious entries
ok        working-tree         ~/.local/share/chezmoi is a directory
ok        dest-dir             ~ is a directory
ok        shell                found /usr/bin/zsh
ok        edit-command         found /usr/bin/vim
ok        umask                022
ok        git-command          found /usr/bin/git, version 2.33.1
ok        merge-command        found /usr/bin/vimdiff
failed    age-command          found /usr/bin/age, could not parse version from (devel)
ok        gpg-command          found /usr/bin/gpg, version 2.2.29
info      pinentry-command     not set
info      1password-command    op not found in $PATH
info      bitwarden-command    bw not found in $PATH
info      gopass-command       gopass not found in $PATH
info      keepassxc-command    keepassxc-cli not found in $PATH
info      keepassxc-db         not set
info      lastpass-command     lpass not found in $PATH
info      pass-command         pass not found in $PATH
info      vault-command        vault not found in $PATH
info      secret-command       not set
```

## Additional context

It's a little weird that age is reporting its version as `devel` in the first place honestly, but this shouldn't be a problem.

**Comment by twpayne** [comment]
Thanks for reporting this.

Yeah, the problem here is that it looks like Arch is not building age correctly. age version 1.0.0 was released some time ago, and 1.0.0-rc1 was around for several months before that.

I think the fix here is to change the doctor result from "failed" to "warning".

**Comment by uhthomas** [comment]
Once again, thanks for a quick turnaround. Changes look good. I'll see if I can open an upstream issue with the ABS community repository for `age`.
